src/xinterpreter.cpp

```diff
@@ -357,7 +362,9 @@ __get_cxx_version ()
 
     void interpreter::init_includes()
     {
-        Cpp::AddIncludePath((xeus::prefix_path() + "/include/").c_str());
+#ifndef EMSCRIPTEN
+         Cpp::AddIncludePath((xeus::prefix_path() + "/include/").c_str());
+#endif
```

Comment:
This would end up giving up `$MAMBA_ROOT_PREFIX/envs/xeus-cpp-wasm-host/include/` which is of no use to us as it's a local path web assembly would just want to ignore. 

But even if we go for it, it would lead to path lookup errors for now.

![image](https://github.com/user-attachments/assets/124765c2-2aaa-4590-b6bf-215d17707817)

Reply:
This can go in `createInterpreter` as `CxxSystemIncludes.push_back(xeus::prefix_path() + "/include/")`

Reply:
Hmm kinda confused as to why this should be treated as a system include rather than using `AddIncludePath`

Do we want to have something like `-isystem /Users/anutosh491/micromamba/envs/xeus-cpp/include/` while building xeus-cpp for non wasm cases ?

Reply:
As we won't be interested in the `init_includes` method as a whole for the wasm case, we could only have it for the non-wasm case at the top itself
```
    interpreter::interpreter(int argc, const char* const* argv) :
       .....
    {
        .....
        init_includes();
        init_preamble();
        init_magic();
    }
```
    
We could only enable it for the non wasm case rather than doing this.
    
```
        void interpreter::init_includes()
    {
#ifndef EMSCRIPTEN
        Cpp::AddIncludePath((xeus::prefix_path() + "/include/").c_str());
#endif
    }
```
    
The same applies for `init_magics` as we have the following (though I strongly believe the file one can be supported in the near future.)
```
    void interpreter::init_magic()
    {
#ifndef EMSCRIPTEN
        preamble_manager["magics"].get_cast<xmagics_manager>().register_magic("xassist", xassist());
        preamble_manager["magics"].get_cast<xmagics_manager>().register_magic("file", writefile());
#endif
    }
```

Reply:
> Hmm kinda confused as to why this should be treated as a system include rather than using `AddIncludePath`
> 
> Do we want to have something like `-isystem /Users/anutosh491/micromamba/envs/xeus-cpp/include/` while building xeus-cpp for non wasm cases ?

Isn’t that folder containing other conda packages?

Reply:
Yes it is but using `AddIncludePath` is enough to add it to the search paths I think.  This is what I see locally !
```
#include "..." search starts here:

#include <...> search starts here:

 /Users/anutosh491/micromamba/envs/xeus-cpp/include

 /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1

 /Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include

 /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include

 /Library/Developer/CommandLineTools/usr/include

 /Users/anutosh491/micromamba/envs/xeus-cpp/bin/../include/c++/v1

 /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks (framework directory)

End of search list.
```

Reply:
My suggestion is to avoid spilling ifdefs all over the place but have it in one.

Reply:
Yes, I get it. Let's do one thing. As we don't expect the init_include, init_magic or init_preamble to work for starters. 

Hence a place we could start with rather than addressing everything separately is this as what I've done in my latest commit.

```
#ifndef EMSCRIPTEN
        init_includes();
        init_preamble();
        init_magic();
#endif
```

And one by one whatever we expect to get working for eg 
1) introspection like `?std::vector`
2) the file magic 

We could start separating those out. And the ones that we don't expect to work for eg `xassist` depends on curl and curl is something that won't be working for wasm as of now .... so these would still be kept to be undefined in case of a wasm build.

Reply:
My doubt is do we get rid of the init_includes method overall and just shift the functionality to createInterpreter ?

Reply:
> This can go in `createInterpreter` as `CxxSystemIncludes.push_back(xeus::prefix_path() + "/include/")`

That' what I am asking in this PR.

Reply:
> That' what I am asking in this PR.

Actually you know what, the `init_includes` doesn't even play a big part as we think it is !!

That's because it's overlapping with the following in kernel.json.in

https://github.com/compiler-research/xeus-cpp/blob/a7649c0e901f4618a6ce60df2155716959346288/share/jupyter/kernels/xcpp20/kernel.json.in#L12-L13

So the location we are interested in is taken care of anyways as I see the following locally coming out of `kernel.json`
```
#include "..." search starts here:
#include <...> search starts here:
 /Users/anutosh491/micromamba/envs/xeus-cpp/include
 /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1
 /Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include
 /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include
 /Library/Developer/CommandLineTools/usr/include
 /Users/anutosh491/micromamba/envs/xeus-cpp/bin/../include/c++/v1
 /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks (framework directory)
End of search list.
```